### PR TITLE
Feat/contract input generation ui

### PR DESCRIPTION
**Pull request by azara**
add wizard for input generation

### CODE REVIEW

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/core/mapping/FieldToContractInputMappingFactory.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+/**
+ * @author aurelie
+ */
+public class FieldToContractInputMappingFactory {
+
+    private final BusinessObjectModelRepositoryStore businessObjectStore;
+
+    public FieldToContractInputMappingFactory(final BusinessObjectModelRepositoryStore businessObjectStore) {
+        this.businessObjectStore = businessObjectStore;
+
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/core/mapping/FieldToContractInputMappingFactory.java]: extra empty line can be removed

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/core/mapping/FieldToContractInputMappingFactory.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            return new SimpleFieldToContractInputMapping((SimpleField) field);
+        } else if (field instanceof RelationField) {
+            return createRelationFieldToContractInputMapping((RelationField) field);
+        }
+        throw new IllegalStateException("Unkwown Field type");
+    }
+
+    private FieldToContractInputMapping createRelationFieldToContractInputMapping(final RelationField field) {
+        final RelationFieldToContractInputMapping relationFieldMapping = new RelationFieldToContractInputMapping(field);
+        if (((RelationField) relationFieldMapping.getField()).getType().equals(Type.COMPOSITION)) {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/core/mapping/FieldToContractInputMappingFactory.java]: invert equals, prefer to put constant first

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/property/ContractContainerAdaptableSelectionProvider.java [hunk 3]
```diff
@@ -36,7 +36,13 @@
...
     public Object getAdapter(final Class adapter) {
         if (EObject.class.equals(adapter)) {
-            return asContractContainer((EObject) ((IAdaptable) unwrap(selection)).getAdapter(EObject.class));
+            final Object unwrapSelection = unwrap(selection);
+            if (unwrapSelection instanceof IAdaptable) {
+                return asContractContainer((EObject) ((IAdaptable) unwrapSelection).getAdapter(EObject.class));
+            } else if (unwrapSelection instanceof EObject) {
+                return asContractContainer((EObject) unwrapSelection);
+            }
+
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/property/ContractContainerAdaptableSelectionProvider.java]: can remove this extra line

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/provider/BusinessObjectFieldsContentProvider.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.eclipse.jface.viewers.IContentProvider#dispose()
+     */
+    @Override
+    public void dispose() {
+        // TODO Auto-generated method stub
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/provider/BusinessObjectFieldsContentProvider.java]: remov eauto-genrated stuff

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/provider/BusinessObjectFieldsContentProvider.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.eclipse.jface.viewers.IContentProvider#inputChanged(org.eclipse.jface.viewers.Viewer, java.lang.Object, java.lang.Object)
+     */
+    @Override
+    public void inputChanged(final Viewer viewer, final Object oldInput, final Object newInput) {
+        // TODO Auto-generated method stub
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/provider/BusinessObjectFieldsContentProvider.java]: remove auto-genrated stuff

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/provider/BusinessObjectFieldsContentProvider.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        return collection.toArray();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.eclipse.jface.viewers.ITreeContentProvider#getChildren(java.lang.Object)
+     */
+    @Override
+    public Object[] getChildren(final Object parentElement) {
+        if (parentElement instanceof RelationField && ((RelationField) parentElement).getType().equals(Type.COMPOSITION)) {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/provider/BusinessObjectFieldsContentProvider.java]: invert equals

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/provider/BusinessObjectFieldsContentProvider.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        return null;
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.eclipse.jface.viewers.ITreeContentProvider#getParent(java.lang.Object)
+     */
+    @Override
+    public Object getParent(final Object element) {
+        // TODO Auto-generated method stub
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/provider/BusinessObjectFieldsContentProvider.java]: remove auto-genrated stuff

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/provider/BusinessObjectFieldsContentProvider.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        return null;
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.eclipse.jface.viewers.ITreeContentProvider#hasChildren(java.lang.Object)
+     */
+    @Override
+    public boolean hasChildren(final Object element) {
+        if (element instanceof RelationField && ((RelationField) element).getType().equals(Type.COMPOSITION)) {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/provider/BusinessObjectFieldsContentProvider.java]: invert equals and maybe factorize, seems to be used in several places

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java [hunk 9]
```diff
@@ -36,8 +98,199 @@ protected CreateContractInputFromBusinessObjectWizardPage() {
...
+        };
+    }
+
+    private void createProcessDataMappingTreeViewer(final Composite composite, final EMFDataBindingContext dbc) {
+        treeViewer = new CheckboxTreeViewer(composite, SWT.FULL_SELECTION | SWT.BORDER | SWT.V_SCROLL);
+        treeViewer.getTree().setLayoutData(GridDataFactory.fillDefaults().grab(true, true).hint(SWT.DEFAULT, 200).create());
+        treeViewer.getTree().setHeaderVisible(true);
+        treeViewer.addCheckStateListener(this);
+        treeViewer.setCheckStateProvider(this);
+        treeViewer.setContentProvider(new ObservableListTreeContentProvider(new IObservableFactory() {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java]: isn't a quite generic ObservableListTreeContentProvider, it migth worth extracting it and/or extracting the classes in arguments

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java [hunk 10]
```diff
@@ -36,8 +98,199 @@ protected CreateContractInputFromBusinessObjectWizardPage() {
...
+    public List<FieldToContractInputMapping> getMappings() {
+        return mappings;
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.eclipse.jface.viewers.ICheckStateListener#checkStateChanged(org.eclipse.jface.viewers.CheckStateChangedEvent)
+     */
+    @Override
+    public void checkStateChanged(final CheckStateChangedEvent event) {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java]: why not crating a specific class implmenting ICheckStateListener?

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java [hunk 11]
```diff
@@ -36,8 +98,199 @@ protected CreateContractInputFromBusinessObjectWizardPage() {
...
+        final FieldToContractInputMapping mapping = (FieldToContractInputMapping) event.getElement();
+        mapping.setGenerated(event.getChecked());
+
+        // SELECT/DESELECT ALL CHILDREN
+        final CheckboxTreeViewer checkboxTreeViewer = (CheckboxTreeViewer) event.getSource();
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java]: if you need comment you need to extract a method

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java [hunk 12]
```diff
@@ -36,8 +98,199 @@ protected CreateContractInputFromBusinessObjectWizardPage() {
...
+        checkboxTreeViewer.setSubtreeChecked(mapping, event.getChecked());
+
+        //SELECT PARENT IF CHILD IS SELECTED
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java]: if you need comment you need to extract a method

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java [hunk 13]
```diff
@@ -36,8 +98,199 @@ protected CreateContractInputFromBusinessObjectWizardPage() {
...
+        if (event.getChecked()) {
+            if (mapping.getParent() != null) {
+                checkboxTreeViewer.setChecked(mapping.getParent(), true);
+            }
+        }
+
+        // DESELECT PARENT IF NO CHILD SELECTED
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java]: if you need comment you need to extract a method

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/labelProvider/FieldNameColumnLabelProvider.java [hunk 14]
```diff
@@ -12,25 +12,27 @@
...
+     * @see org.eclipse.jface.viewers.ColumnLabelProvider#getText(java.lang.Object)
+     */
+    @Override
+    public String getText(final Object element) {
 
-    public AddContractInputFromMappingCommand(final EditingDomain domain, final Contract contract, final List<FieldToContractInputMapping> mappings) {
-        for (final FieldToContractInputMapping mapping : mappings) {
-            append(AddCommand.create(domain, contract, ProcessPackage.Literals.CONTRACT__INPUTS, mapping.toContractInput()));
+        if (element instanceof FieldToContractInputMapping) {
+            return ((FieldToContractInputMapping) element).getField().getName();
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/labelProvider/FieldNameColumnLabelProvider.java]: the field cannot be null?

**Comment by rbioteau** [comment]
Please remake PR on CP3

**Pull request closed without merging**